File: app/src/main/java/com/iita/akilimo/utils/enums/EnumScrollFlags.java
package com.iita.akilimo.utils.enums;

import com.google.android.material.appbar.AppBarLayout.LayoutParams;

public enum EnumScrollFlags {
    NO_SCROLL {
        @Override
        public int scrollFlags() {
            return LayoutParams.SCROLL_FLAG_ENTER_ALWAYS;
        }
    },
    SCROLL {
        @Override
        public int scrollFlags() {
            return LayoutParams.SCROLL_FLAG_SCROLL | LayoutParams.SCROLL_FLAG_EXIT_UNTIL_COLLAPSED;
        }
    },
    ;

    public abstract int scrollFlags();
}

File: app/src/main/java/com/akilimo/mobile/utils/enums/EnumScrollFlags.kt
package com.akilimo.mobile.utils.enums

import com.google.android.material.appbar.AppBarLayout

enum class EnumScrollFlags {
    NO_SCROLL {
        override fun scrollFlags(): Int {
            return AppBarLayout.LayoutParams.SCROLL_FLAG_ENTER_ALWAYS
        }
    },
    SCROLL {
        override fun scrollFlags(): Int {
            return AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL or AppBarLayout.LayoutParams.SCROLL_FLAG_EXIT_UNTIL_COLLAPSED
        }
    };

    abstract fun scrollFlags(): Int
}
